cmake/modules/AddSwift.cmake

```diff
@@ -2326,44 +2326,47 @@ macro(add_swift_lib_subdirectory name)
 endmacro()
 
 function(add_swift_host_tool executable)
-  set(ADDSWIFTHOSTTOOL_multiple_parameter_options
-        SWIFT_COMPONENT
+  set(options)
+  set(single_parameter_options)
+  set(multiple_parameter_options
         COMPILE_FLAGS
         DEPENDS
+        SWIFT_COMPONENT
         SWIFT_MODULE_DEPENDS)
 
-  cmake_parse_arguments(
-      ADDSWIFTHOSTTOOL # prefix
-      "" # options
-      "" # single-value args
-      "${ADDSWIFTHOSTTOOL_multiple_parameter_options}" # multi-value args
-      ${ARGN})
-
-  # Configure variables for this subdirectory.
-  set(MODULE_VARIANT_SUFFIX "-swiftmodule${SWIFT_HOST_VARIANT_SUFFIX}")
+  cmake_parse_arguments(ASHT
+                        "${options}"
+                        "${single_parameter_options}"
+                        "${multiple_parameter_options}"
+                        ${ARGN})
 
-  foreach(mod ${ADDSWIFTHOSTTOOL_SWIFT_MODULE_DEPENDS})
-    list(APPEND ADDSWIFTHOSTTOOL_DEPENDS "swift${mod}${MODULE_VARIANT_SUFFIX}")
-    list(APPEND ADDSWIFTHOSTTOOL_DEPENDS "swift${mod}${SWIFT_HOST_VARIANT_SUFFIX}")
+  foreach(mod ${ASHT_SWIFT_MODULE_DEPENDS})
+    list(APPEND ASHT_DEPENDS
+         "swift${mod}-swiftmodule${SWIFT_HOST_VARIANT_SUFFIX}"
+         "swift${mod}${SWIFT_HOST_VARIANT_SUFFIX}")
   endforeach()
 
   # Create the executable rule.
-  add_swift_executable(
-    ${executable} 
-    ${ADDSWIFTHOSTTOOL_UNPARSED_ARGUMENTS}
-    DEPENDS ${ADDSWIFTHOSTTOOL_DEPENDS}
-    COMPILE_FLAGS ${ADDSWIFTHOSTTOOL_COMPILE_FLAGS}
-  )
+  add_swift_executable(${executable}
+                       ${ASHT_UNPARSED_ARGUMENTS}
+                       DEPENDS ${ASHT_DEPENDS}
+                       COMPILE_FLAGS ${ASHT_COMPILE_FLAGS})
+  llvm_externalize_debuginfo(${executable})
```

Comment:
What used to happen without externalizing the debug info/codesign? Can you add a comment here to explain why we are doing this here?

My main concern as always is that people who work in swift may not be 100% familiar with how LLVM's cmake works (I imagine that over time that will change if we get the host side completely compiling with pure LLVM).

Reply:
Without externalizing the debug info, we would keep the debug info in the binary until the install time.  By immediately externalizing, we generate a dSYM bundle on macOS and the following codesign step signs the binaries (which need to be stripped previously because you do not want the debug info to be included in the signature).  I can add a comment about that.

Reply:
@compnerd I guess what we did previously was IIRC there is an additional step in build-script-impl that created the dSYM, no?